pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -1334,6 +1334,14 @@ func updatePipelineRunStatusFromChildObjects(ctx context.Context, logger *zap.Su
 	fullEmbedded := cfg.FeatureFlags.EmbeddedStatus == config.FullEmbeddedStatus || cfg.FeatureFlags.EmbeddedStatus == config.BothEmbeddedStatus
 	minimalEmbedded := cfg.FeatureFlags.EmbeddedStatus == config.MinimalEmbeddedStatus || cfg.FeatureFlags.EmbeddedStatus == config.BothEmbeddedStatus
 
+	if cfg.FeatureFlags.EmbeddedStatus == config.MinimalEmbeddedStatus {
```

Comment:
I think this set of conditionals will be more readable with a `switch` statement, since some of them overlap.
e.g.

```
switch cfg.FeatureFlags.EmbeddedStatus {
  case  config.MinimalEmbeddedStatus:
    // clear taskruns and runs
   // add childrefs
  case config.FullEmbeddedStatus:
    // add taskruns and runs
   // clear childrefs
  case config.BothEmbeddedStatus:
   // add both
}
```